node/tests/SharedTests.ts

```diff
@@ -2527,6 +2527,149 @@ export function runBaseTests<Context>(config: {
         },
         config.timeout,
     );
+
+    it.each([ProtocolVersion.RESP2, ProtocolVersion.RESP3])(
+        "object encoding test_%p",
```

Comment:
This test is based on Aaron's Python test

Reply:
Which is based on Jonathan's java test